.github/workflows/ci.yml

```diff
@@ -0,0 +1,123 @@
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - v*
+  pull_request:
+
+jobs:
+  shellcheck:
+    name: Shellcheck
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run ShellCheck
+        run: |
+          sudo apt-get install shellcheck
+          shopt -s globstar; shellcheck script/**/*.sh
+
+  build:
+    name: ${{ matrix.friendlyName }} ${{ matrix.arch }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-10.15, windows-2019, ubuntu-18.04]
+        arch: [32, 64]
+        include:
+          - os: macos-10.15
+            friendlyName: macOS
+            targetPlatform: macOS
+          - os: windows-2019
+            friendlyName: Windows
+            targetPlatform: win32
+          - os: ubuntu-18.04
+            friendlyName: Linux
+            targetPlatform: ubuntu
+        exclude:
+          - os: macos-10.15
+            arch: 32
+          - os: ubuntu-18.04
+            arch: 32
+    timeout-minutes: 20
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+          # Needed for script/package.sh to work
+          fetch-depth: 0
+      - name: Install dependencies
+        run: npm install
+      - name: Check formatting
+        run: npm run prettier
+      - name: Build tools
+        run: npm run check
+      - name: Install extra dependencies for building Git on Ubuntu
+        if: matrix.targetPlatform == 'ubuntu'
+        run: sudo apt-get install libcurl4-openssl-dev libexpat1-dev gettext
+      - name: Build
+        shell: bash
+        run: script/build.sh
+        env:
+          TARGET_PLATFORM: ${{ matrix.targetPlatform }}
+          TARGET_ARCH: ${{ matrix.arch }}
+      - name: Package
+        shell: bash
+        run: script/package.sh
+        env:
+          TARGET_PLATFORM: ${{ matrix.targetPlatform }}
+          TARGET_ARCH: ${{ matrix.arch }}
+      - name: Upload ouptut artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name:
+            dugite-native-${{ matrix.targetPlatform }}-${{ matrix.arch }}-output
+          path: ./output
+          retention-days: 5
+
+  release:
+    name: Create GitHub release
+    needs: build
+    if: github.event_name == 'push' && contains(github.ref, 'refs/tags/')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Download all artifacts
+        uses: actions/download-artifact@v2
+        with:
+          path: './artifacts'
+
+      - name: Display structure of downloaded files
+        run: ls -R
+        working-directory: './artifacts'
+
+      - name: Create release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ github.ref }}
+          release_name: Git ${{ github.ref }}
+          body: |
+            TODO add release notes
```

Comment:
Could add `generate-release-notes.ts` here, what do you think?

Reply:
I think we should rewrite `generate-release-notes.ts` to work exclusively on the artifacts from the build job instead of caring about the GitHub release. I don't see much value (other than it being less work probably) of having it operate on the release using the API.

Let me know if you want me to work on rewriting the script, you've already put so much effort into this so I can understand if you don't want to rewrite our legacy scripts as well 😛

Reply:
@niik how does this sound: if I make a start on rewriting the script, would you mind having a quick look at https://github.com/desktop/desktop/pull/9691#issuecomment-727833940? That would make me and many others very happy 😉